### PR TITLE
[Feature] Add `created_at` and `updated at` fields

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** [#144 in dbt_quickbooks](https://github.com/fivetran/dbt_quickbooks/issues/144)

**This PR will result in the following new package version:**

Added fields into various models, but nothing in particular will break for the customer I believe. 

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Feature Updates
- We introduced these pre-existing timestamp fields from these sources to their equivalent staging models to better analyze real-time transaction data: 
  - `created_at`: `bill`, `bill_payment`, `credit_memo`, `invoice`, `payment`, `transfer`
  - `updated_at`: `bill`, `bill_payment`, `credit_memo`, `deposit`, `invoice`, `journal_entry`, `payment`, `purchase`, `refund_receipt`, `sales_receipt`, `transfer`, `vendor_credit`
- These new fields are then being brought in downstream in the `dbt_quickbooks` package in the `quickbooks__general_ledger` model via the double entry transaction intermediate models. You can learn more about these changes in the [v0.17.0 release of the `dbt_quickbooks` package](https://github.com/fivetran/dbt_quickbooks/releases/tag/v0.17.0). 

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [NA] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Ensured the `created_at` and `updated_at` fields were being brought into the relevant models that were used in the double entry transaction models. Example:
Screenshot 2024-11-20 at 6 03 54 AM

### If you had to summarize this PR in an emoji, which would it be?
 
⌛

### CODE REVIEW

File: models/stg_quickbooks.yml [hunk 1]
```diff
@@ -826,6 +844,8 @@ models:
         description: The value of this field will always be set to zero.
       - name: transaction_date
         description: Date when the journal entry was transacted.
+      - name: updated_at
+        description: Timestamp of the date which the journal entry was last updated.
```
Comment (fivetran-joemarkiewicz) [inline models/stg_quickbooks.yml:848]: Since we're making this update I noticed the `source_relation` is not documented for Journal Entry. Can you add this to the documentation.
image

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
Thanks for these updates @fivetran-avinash, just a few change requests and a small question before approval.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
+# dbt_quickbooks_source v0.11.1
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply CHANGELOG.md]: Updated

File: dbt_project.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 config-version: 2
 name: 'quickbooks_source'
 
-version: '0.11.0'
+version: '0.11.1'
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply dbt_project.yml]: Bumped

File: integration_tests/dbt_project.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name: 'quickbooks_source_integration_tests'
-version: '0.11.0'
+version: '0.11.1'
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply integration_tests/dbt_project.yml]: Bumped

File: models/stg_quickbooks.yml (hunk 1 above)
Comment (fivetran-avinash) [reply models/stg_quickbooks.yml:850]: Added

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz This is ready for re-review.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,13 @@
+# dbt_quickbooks_source v0.12.0
+[PR #58](https://github.com/fivetran/dbt_quickbooks_source/pull/57) introduces the following updates:
+
+## Breaking Changes
+- We introduced these pre-existing timestamp fields from these sources to their equivalent staging models to better analyze real-time transaction data:
```
Comment (fivetran-catfritz) [inline CHANGELOG.md:5]: I would list the field and the exact staging models they were added to instead of requiring the user to have to match them up on their own. 

```suggestion
- Introduced the following timestamp fields to the listed staging models to better analyze real-time transaction data:
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,13 @@
+# dbt_quickbooks_source v0.12.0
+[PR #58](https://github.com/fivetran/dbt_quickbooks_source/pull/57) introduces the following updates:
+
+## Breaking Changes
+- We introduced these pre-existing timestamp fields from these sources to their equivalent staging models to better analyze real-time transaction data: 
+  - `created_at`: `bill`, `bill_payment`, `credit_memo`, `invoice`, `payment`, `transfer`
+  - `updated_at`: `bill`, `bill_payment`, `credit_memo`, `deposit`, `invoice`, `journal_entry`, `payment`, `purchase`, `refund_receipt`, `sales_receipt`, `transfer`, `vendor_credit`
+- These new fields are then being brought in downstream in the `dbt_quickbooks` package in the `quickbooks__general_ledger` model via the double entry transaction intermediate models. You can learn more about these changes in the [v0.17.0 release of the `dbt_quickbooks` package](https://github.com/fivetran/dbt_quickbooks/releases/tag/v0.17.0). 
+- As this introduces new columns to our staging models and changes our schema, this will be a breaking change.
```
Comment (fivetran-catfritz) [inline CHANGELOG.md:9]:
```suggestion
- As this introduces new columns to our staging models and changes our schema, this is breaking change. 
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,13 @@
+# dbt_quickbooks_source v0.12.0
+[PR #58](https://github.com/fivetran/dbt_quickbooks_source/pull/57) introduces the following updates:
+
+## Breaking Changes
+- We introduced these pre-existing timestamp fields from these sources to their equivalent staging models to better analyze real-time transaction data: 
+  - `created_at`: `bill`, `bill_payment`, `credit_memo`, `invoice`, `payment`, `transfer`
+  - `updated_at`: `bill`, `bill_payment`, `credit_memo`, `deposit`, `invoice`, `journal_entry`, `payment`, `purchase`, `refund_receipt`, `sales_receipt`, `transfer`, `vendor_credit`
+- These new fields are then being brought in downstream in the `dbt_quickbooks` package in the `quickbooks__general_ledger` model via the double entry transaction intermediate models. You can learn more about these changes in the [v0.17.0 release of the `dbt_quickbooks` package](https://github.com/fivetran/dbt_quickbooks/releases/tag/v0.17.0).
```
Comment (fivetran-catfritz) [inline CHANGELOG.md:8]:
```suggestion
- These new fields are incorporated in the `dbt_quickbooks` package's `quickbooks__general_ledger` model via the `*_double_entry` intermediate models. You can learn more about these changes in the [v0.17.0 release of the `dbt_quickbooks` package](https://github.com/fivetran/dbt_quickbooks/releases/tag/v0.17.0). 
```

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-avinash a couple suggestions!

File: CHANGELOG.md (hunk 5 above)
Comment (fivetran-avinash) [reply CHANGELOG.md:5]: @fivetran-catfritz Listing all the staging models would get a bit wordy, so I did a slight tweak to your suggestion. Let me know your thoughts.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-catfritz Addressed your comments!

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,13 @@
+# dbt_quickbooks_source v0.12.0
+[PR #58](https://github.com/fivetran/dbt_quickbooks_source/pull/57) introduces the following updates:
+
+## Breaking Changes
+- Introduced the following timestamp fields to the listed `stg_quickbooks__*` models to better analyze real-time transaction data:
+  - `created_at`: `bill`, `bill_payment`, `credit_memo`, `invoice`, `payment`, `transfer`
+  - `updated_at`: `bill`, `bill_payment`, `credit_memo`, `deposit`, `invoice`, `journal_entry`, `payment`, `purchase`, `refund_receipt`, `sales_receipt`, `transfer`, `vendor_credit`
+- - These new fields are incorporated in the `dbt_quickbooks` package's `quickbooks__general_ledger` model via the `*_double_entry` intermediate models. You can learn more about these changes in the [v0.17.0 release of the `dbt_quickbooks` package](https://github.com/fivetran/dbt_quickbooks/releases/tag/v0.17.0).
```
Comment (fivetran-catfritz) [inline CHANGELOG.md:8]: Just noticed the double dash.
```suggestion
- These new fields are incorporated in the `dbt_quickbooks` package's `quickbooks__general_ledger` model via the `*_double_entry` intermediate models. You can learn more about these changes in the [v0.17.0 release of the `dbt_quickbooks` package](https://github.com/fivetran/dbt_quickbooks/releases/tag/v0.17.0).
```

File: CHANGELOG.md (hunk 5 above)
Comment (fivetran-catfritz) [reply CHANGELOG.md]: True--this makes sense!

**Review by fivetran-catfritz — APPROVED** [review]
@fivetran-avinash One more teensy item,  but lgtm!
